test/ocamlorg_web/graphql_test.ml

```diff
@@ -0,0 +1,201 @@
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "Decentralised installation system - GTK UI"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "abt")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "Package dependency solve"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "ocaml")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "OCaml port of CMU's abstract binding trees"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "absolute")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis =
+          "Interactive theorem prover based on lambda-tree syntax"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ]
+
+let start_with_test str1 str2 cond () =
+  Alcotest.(check bool)
+    "is true or false"
+    cond
+    (Ocamlorg_web.Graphql.starts_with str1 str2)
+
+let is_package_test name1 name2 cond () =
+  Alcotest.(check bool)
+    "is false"
+    cond
+    (Ocamlorg_web.Graphql.is_package name1 name2)
+
+let get_packages_result_test () =
+  let total_packages = List.length packages in
+  let offset = 0 in
+  let limit = 3 in
+  let filter = None in
+  let all_packages =
+    Ocamlorg_web.Graphql.get_packages_result
+      total_packages
+      offset
+      limit
+      filter
+      packages
+  in
+  let num_of_pakgs = List.length all_packages.packages in
+  Alcotest.(check int) "returns 3 packages" 3 num_of_pakgs
+
+let get_deps_test () =
+  let deps = Ocamlorg_web.Graphql.get_info dependencies in
+  let num_of_deps = List.length deps in
+  Alcotest.(check int) "returns 6 dependencies" 6 num_of_deps
+
+let get_single_package_test name cond () =
+  let pkg =
+    List.find_opt
+      (fun package ->
+        Ocamlorg_web.Graphql.is_package
+          name
+          (Package.Name.to_string (Package.name package)))
+      packages
+  in
+  let res = match pkg with None -> false | Some _ -> true in
+  Alcotest.(check bool) "ocaml exist in package list" cond res
```

Comment:
You've copied [the resolve function](https://github.com/ocaml/v3.ocaml.org-server/pull/60/files#diff-7b9c15d336b381008174d3a481ce2b8f7812523a2ee852caf532f4fa332857feR290-R293) for the query by package name here. The problem about copying code from your implementation in your test is that if there's a problem in the logic of your implementation, you won't capture that in the test.

Since you're already testing the function `Ocamlorg_web.Graphql.is_package` above, I assume that here you'd like to test the whole [resolve function](https://github.com/ocaml/v3.ocaml.org-server/pull/60/files#diff-7b9c15d336b381008174d3a481ce2b8f7812523a2ee852caf532f4fa332857feR288-R293) for the query by package name. Is that correct? If so, I'd suggest that you extract the [logic of the resolve function](https://github.com/ocaml/v3.ocaml.org-server/pull/60/files#diff-7b9c15d336b381008174d3a481ce2b8f7812523a2ee852caf532f4fa332857feR290-R293) in `graphql.ml` into a separate function and expose that one in `graphql.mli`. Then you can call that function directly here on your mock data `packages` and on your test parameter `name`. What do you think?

Reply:
Done, thank you for pointing that out

Reply:
Thanks!